File: SPRING/Hospital_Management_system_angular/src/app/services/emergency/triage.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { Triage } from '../../models/emergency';

@Injectable({ providedIn: 'root' })
export class TriageService {
  private api = environment.apiUrl + 'emergency/triage';

  constructor(private http: HttpClient) {}

  create(triage: Triage): Observable<Triage> {
    return this.http.post<Triage>(this.api, triage);
  }

  getByEmergencyPatientId(emergencyPatientId: number): Observable<Triage> {
    return this.http.get<Triage>(`${this.api}/patient/${emergencyPatientId}`);
  }

  getAll(): Observable<Triage[]> {
    return this.http.get<Triage[]>(this.api);
  }

  update(id: number, triage: Triage): Observable<Triage> {
    return this.http.put<Triage>(`${this.api}/${id}`, triage);
  }

  getTriageDistribution(): Observable<any> {
    return this.http.get<any>(`${this.api}/distribution`);
  }
}

File: SPRING/Modern_Hospital_Management_System/src/main/java/emranhss/com/Modern_Hospital_Management_System/service/TriageService.java
package emranhss.com.Modern_Hospital_Management_System.service;

import emranhss.com.Modern_Hospital_Management_System.dto.request.TriageRequest;
import emranhss.com.Modern_Hospital_Management_System.dto.response.TriageResponse;

import java.util.List;
import java.util.Map;

public interface TriageService {

    TriageResponse create(TriageRequest request);

    TriageResponse getByEmergencyPatientId(Long emergencyPatientId);

    List<TriageResponse> getAll();

    TriageResponse update(Long id, TriageRequest request);

    Map<String, Object> getTriageDistribution();
}
